components/database/mariadb-101/Makefile

```diff
@@ -0,0 +1,106 @@
+
+CMAKE= cmake
+
+COMPONENT_NAME=		mariadb
+COMPONENT_VERSION=	10.1.17
+IPS_COMPONENT_VERSION=	10.1.17
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_PROJECT_URL=	http://www.mariadb.org/
+COMPONENT_ARCHIVE_HASH=	sha256:4ca45ac5e34418761868115ebc8c068d511fed08e283b2cac52559d63ba4aab5
+COMPONENT_ARCHIVE_URL=	http://mirrors.coreix.net/$(COMPONENT_NAME)/$(COMPONENT_SRC)/source/$(COMPONENT_ARCHIVE)
+
+include $(WS_TOP)/make-rules/prep.mk
+include $(WS_TOP)/make-rules/configure.mk
+include $(WS_TOP)/make-rules/ips.mk
+
+
+CONFIGURE_PREFIX=/usr/mariadb/10.1
+CONFDIR=/etc/mariadb/10.1
+DATA_PREFIX=/var/mariadb/10.1
+
+CMAKE_BINDIR.32 =	bin
+CMAKE_BINDIR.64 =	bin/$(MACH64)
+CMAKE_LIBDIR.32 =	lib
+CMAKE_LIBDIR.64 =	lib/$(MACH64)
+CMAKE_PLUGINDIR.32 =	lib/plugin
+CMAKE_PLUGINDIR.64 =	lib/$(MACH64)/plugin
+
+CMAKE_OPTIONS+= -DCMAKE_C_COMPILER=$(CC)
+CMAKE_OPTIONS+= -DCMAKE_CXX_COMPILER=$(CXX)
+CMAKE_OPTIONS+= -DCMAKE_C_FLAGS=$(CFLAGS)
+CMAKE_OPTIONS+= -DCMAKE_CXX_FLAGS=$(CFLAGS)
+
+CMAKE_OPTIONS+= -DINSTALL_LAYOUT=SVR4
+CMAKE_OPTIONS+= -DCMAKE_INSTALL_PREFIX=$(CONFIGURE_PREFIX)
+CMAKE_OPTIONS+= -DINSTALL_BINDIR=$(CMAKE_BINDIR.$(BITS))
+CMAKE_OPTIONS+= -DINSTALL_INFODIR=bin
+CMAKE_OPTIONS+= -DINSTALL_INCLUDEDIR=include
+CMAKE_OPTIONS+= -DINSTALL_LIBDIR=$(CMAKE_LIBDIR.$(BITS))
+CMAKE_OPTIONS+= -DINSTALL_PLUGINDIR=$(CMAKE_PLUGINDIR.$(BITS))
+CMAKE_OPTIONS+= -DINSTALL_MANDIR=man
+CMAKE_OPTIONS+= -DINSTALL_SHAREDIR=share
+CMAKE_OPTIONS+= -DINSTALL_MYSQLSHAREDIR=share
+CMAKE_OPTIONS+= -DINSTALL_SBINDIR=$(CMAKE_BINDIR.$(BITS))
+CMAKE_OPTIONS+= -DINSTALL_SCRIPTDIR=$(CMAKE_BINDIR.$(BITS))
+CMAKE_OPTIONS+= -DMYSQL_DATADIR=$(DATA_PREFIX)/data
+CMAKE_OPTIONS+= -DSYSCONFDIR=$(CONFDIR)
+CMAKE_OPTIONS+= -DINSTALL_SYSCONFDIR=$(CONFDIR)
+CMAKE_OPTIONS+= -DENABLE_DTRACE=off
+CMAKE_OPTIONS+= -DWITH_READLINE=ON
+CMAKE_OPTIONS+= -DWITH_EMBEDDED_SERVER=ON
+CMAKE_OPTIONS+= -DWITHOUT_MROONGA_STORAGE_ENGINE=ON
+CMAKE_OPTIONS+= -DHAVE_VISIBILITY_HIDDEN=0
+CMAKE_OPTIONS+= -DMY_CONTEXT_USE_UCONTEXT=1
+CMAKE_OPTIONS+= -DWITH_EXTRA_CHARSETS=complex
+CMAKE_OPTIONS+= -DWITH_ZLIB=system
+CMAKE_OPTIONS+= -DWITH_PCRE=system
+CMAKE_OPTIONS+= -DWITH_SSL=system
+CMAKE_OPTIONS+= -DINSTALL_SUPPORTFILESDIR=share/mysql
+
+# gcc invocation in some link.txt file is missing -m$(BITS) parameter
+# and it makes 64-bit build to fail, this is a quick hack to make it build
+# for now, cmake build system needs a proper fix
+
+COMPONENT_POST_CONFIGURE_ACTION.32=
+COMPONENT_POST_CONFIGURE_ACTION.64= find $(@D) -type f -name link.txt \
+    -exec perl -pi -e 's/gcc\ \ -fPIC/\/gcc\ -m64\ -fPIC/' {} \;
+COMPONENT_POST_CONFIGURE_ACTION=$(COMPONENT_POST_CONFIGURE_ACTION.$(BITS))
+
+$(BUILD_DIR)/%/.configured:	$(SOURCE_DIR)/.prep
+	($(RM) -rf $(@D) ; $(MKDIR) $(@D))
+	$(COMPONENT_PRE_CONFIGURE_ACTION)
+	(cd $(@D) ; $(ENV) $(CONFIGURE_ENV) $(CMAKE) $(CMAKE_OPTIONS) $(SOURCE_DIR))
+	$(COMPONENT_POST_CONFIGURE_ACTION)
+	$(TOUCH) $@
```

Comment:
When initial component was created there was no cmake.mk. Can we switch component to using cmake.mk? This should make such workarounds not necessary.